backend/hitas/services/thirty_year_regulation.py

```diff
@@ -125,23 +128,30 @@ def perform_thirty_year_regulation(calculation_date: datetime.date) -> Regulatio
         f"{len(results['skipped'])} could not be checked."
     )
 
+    housing_companies += split_housing_companies
+
+    logger.info("Updating housing company states...")
     _free_housing_companies_from_regulation(housing_companies, results)
 
     logger.info("Regulation check complete!")
     return results
 
 
-def _check_automatic_release(housing_companies: list[HousingCompanyWithAnnotations]) -> list[ComparisonData]:
+def _split_automatically_released(
+    housing_companies: list[HousingCompanyWithAnnotations],
+) -> tuple[list[HousingCompanyWithAnnotations], list[ComparisonData]]:
     """
     Separate out housing companies that are automatically released from regulation.
     """
+    split_housing_companies: list[HousingCompanyWithAnnotations] = []
     automatically_released: list[ComparisonData] = []
```

Comment:
They both contain the automatically released housing companies, the `split_housing_companies` just contains the split housing company itself (so that they are not included in the rest of the regulation, but their state can be updated at the end), and `automatically_released` contains the comparison data for the `automatically_released` key in the `RegulationResults` given from the API.